Bin/Snapshot.php

```diff
@@ -273,16 +316,50 @@ function () use ($tags, $newTag, $repositoryRoot, &$changelog) {
                             $title = $newTag;
                         }
 
-                        $changelog .=
-                            '# ' . $title . "\n\n" .
-                            Console\Processus::execute(
-                                'git --git-dir=' . $repositoryRoot . '/.git ' .
-                                    'log ' .
-                                        '--first-parent ' .
-                                        '--pretty="format:  * %s (%aN, %aI)" ' .
-                                        $fromStep . '...' . $toStep,
-                                false
-                            ) . "\n\n";
+                        $changelog .= '# ' . $title . "\n\n";
+
+                        $commits = explode("\n", Console\Processus::execute(
+                            'git --git-dir=' . $repositoryRoot . '/.git ' .
+                                'log ' .
+                                    '--first-parent ' .
+                                    '--pretty="format:  * %s (%aN, %aI)" ' .
+                                    $fromStep . '...' . $toStep,
+                            false
+                        ));
+
+                        $typedCommits = [];
+                        $untypedCommits = [];
+                        foreach ($commits as $commit) {
+                            if (preg_match($changelogThemesRegex, trim($commit), $matches)) {
+                                if (!isset($typedCommits[$matches['type']])) {
+                                    $typedCommits[$matches['type']] = [$matches['section'] => []];
+                                } elseif (!isset($typedCommits[$matches['type']][$matches['section']])) {
+                                    $typedCommits[$matches['type']][$matches['section']] = [];
+                                }
+
+                                $typedCommits[$matches['type']][$matches['section']][] = ' *' . $matches['message'];
```

Comment:
In order to keep responsabilities separate, I would remove the `'  * '` from here, and add it in the next loop.

Reply:
Ok, updated.